server/workflowarchive/archived_workflow_server.go

```diff
@@ -298,13 +302,20 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 				return nil, sutils.ToStatusError(err, codes.Internal)
 			}
 		}
-
+		log.WithFields(log.Fields{"Dehydrate workflow uid=": wf.UID}).Info("RetryArchivedWorkflow")
+		err = w.hydrator.Dehydrate(wf)
+		if err != nil {
+			return nil, sutils.ToStatusError(err, codes.Internal)
+		}
+		
 		wf.ObjectMeta.ResourceVersion = ""
 		wf.ObjectMeta.UID = ""
 		result, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Create(ctx, wf, metav1.CreateOptions{})
 		if err != nil {
 			return nil, sutils.ToStatusError(err, codes.Internal)
 		}
+		offloadVersion, _ := w.offloadNodeStatusRepo.Get(string(oriUid), wf.GetOffloadNodeStatusVersion())
```

Comment:
can you change "offloadVersion" to "offloadedNodes" and 
do some error handling？ thanks！

Reply:
> can you change "offloadVersion" to "offloadedNodes" and do some error handling？ thanks！

yeah, I have rename the args and handle the errors. @shuangkun